.github/workflows/checks.yml

```diff
@@ -3,27 +3,282 @@ name: Checks
-          # add requirements
-          sudo apt update; sudo apt install portaudio19-dev gettext rename
+          sudo apt update
+          sudo apt install portaudio19-dev gettext rename
           sudo snap install ripgrep --classic
-          export CARGO_TARGET_DIR=~/target
-          export RSPY_TARGET_DIR=~/target
-          make check build BUILDFLAGS=""
+      - run: make develop
+      - run: make build
+      - run: make check
+      - name: Upload python wheels
+        uses: actions/upload-artifact@v1
+        with:
+          name: anki_linux_python_wheels
+          path: dist
+
+
+  mac_os_tests:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Configure environment variables
+        run: |
+          # echo "::set-env name=SHELLFLAGS::-x"
+          echo "::set-env name=BUILDFLAGS::"
+          echo "::set-env name=RSPY_TARGET_DIR::${GITHUB_WORKSPACE}/target"
+          echo "::set-env name=CARGO_TARGET_DIR::${GITHUB_WORKSPACE}/target"
+      # Necessary for now for the cargo cache:
+      # https://github.com/actions/cache/issues/133#issuecomment-599102035
+      - name: Fix ~/.cache permissions
+        run: |
+          sudo chown -R $(whoami):$(id -ng) ~/.cargo/
+      - name: Cache cargo registry
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/registry
+          key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}-5-
+      - name: Cache cargo index
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/git
+          key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}-5-
+      - name: Cache cargo target
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/target
+          key: ${{ runner.os }}-cargo-target-${{ hashFiles('**/Cargo.toml') }}-5-
+      - name: Cache cargo rslib
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/rslib/target
+          key: ${{ runner.os }}-cargo-rslib-${{ hashFiles('**/Cargo.toml') }}-5-
+      - name: Cache cargo rspy
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/rspy/target
+          key: ${{ runner.os }}-cargo-rspy-${{ hashFiles('**/Cargo.toml') }}-5-
+      - name: Cache pip wheels
+        uses: actions/cache@v1
+        with:
+          path: ~/Library/Caches/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}-5-
+      - name: Cache pyenv
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/pyenv
+          key: ${{ runner.os }}-pyenv-${{ hashFiles('**/Makefile') }}-${{ hashFiles('**/requirements.txt') }}-5-
+      - name: Set up python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - name: Set up protoc
+        uses: arduino/setup-protoc@v1.1.0
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up node
```

Comment:
Would it be possible to deduplicate some of this code by using a job matrix?

Reply:
I will look into it and see how I can remove the duplicated code.

Reply:
I just did it with my last commit.